src/components/componentDiff/ComponentDiffTab.js

```diff
@@ -212,13 +328,135 @@ const ComponentVersionDiff = ({
   pullRequest,
   deleteDiff,
 }) => {
+  const [openDialog, setOpenDialog] = useState(false)
+  const [excludedItems, setExcludedItems] = useState({ excludedComponents: [], excludedResources: [] })
   const [diff, isDiffLoading, isDiffError] = useFetchCompDiff({
     leftName: leftName,
     rightName: rightName,
     leftVersion: leftVersion,
     rightVersion: rightVersion,
   })
 
+  const handleOpenDialog = () => {
+    setOpenDialog(true)
+  }
+
+  const handleCloseDialog = () => {
+    setOpenDialog(false)
+  }
+
+  const handleExclude = ({ excludedComponents, excludedResources }) => {
+    console.log(excludedComponents, excludedResources)
+    setExcludedItems(prevState => {
+      const newState = {
+        ...prevState,
+        excludedComponents,
+        excludedResources,
+      }
+      handleExportDiff(newState)
+
+      return newState
+    })
+  }
+
+  useEffect(() => {
+  }, [excludedItems])
+
+  const determineMainSourceForComponent = (component, absentOk = true) => {
+    if (!component.sources || component.sources.length === 0) {
+      return null
+    }
+    const mainSource = component.sources.find(source => {
+      const label = source.labels?.find(l => l.name === knownLabelNames.mainSource)
+      return label && label.value && label.value['repository-classification'] === 'main'
+    })
+
+    if (mainSource) {
+      return mainSource.access?.repoUrl || null
+    }
+
+    if (absentOk) {
```

Comment:
`absentOk` should be considered in call cases where null could be returned.
function semantics should be
"""
heuristically determines "main" source for given component
returns null if no main source could be identified, raises instead if not absentOk
"""

This is a means to prepare this function for reusability.